cetlvast/suites/unittest/CMakeLists.txt

```diff
@@ -86,19 +190,15 @@ set_directory_properties(PROPERTIES
     IN_BUILD_TESTS "${ALL_TESTS_RUN}"
 )
 
-if (CMAKE_BUILD_TYPE STREQUAL "Coverage")
-    enable_coverage_report(COVERAGE_REPORT_FORMATS html sonarqube
-                           ROOT_DIRECTORY ${CETL_ROOT}
-                           OUT_REPORT_INDICES LOCAL_COVERAGE_REPORT_INDICIES
-    )
-endif()
+enable_coverage_report(COVERAGE_REPORT_FORMATS html sonarqube
+                        ROOT_DIRECTORY ${CETL_ROOT}
+                        OUT_REPORT_INDICES LOCAL_COVERAGE_REPORT_INDICIES
+)
```

Comment:
I ran "configure-gcc-native-cpp-14-offline - build-Coverage-gcc-native-cpp-14-offline" and opened `build/suites/unittest/gcovr_html/coverage.html`, and noticed quite bad % of Branches coverage.
![image](https://github.com/user-attachments/assets/9ed2b2ae-2f61-477c-be9f-dff015ba976c)
I believe this is heavily affected by our `CETL_DEBUG_ASSERT`, like here:
![image](https://github.com/user-attachments/assets/8e8cea2b-7ecb-4a04-a9cb-f6d40eb99b11)

I propose we suppress CETL_DEBUG_ASSERT for Coverage builds.

Reply:
Should we not just try to get the coverage numbers up over time?

Reply:
Numbers are already good (see **Lines** and even **Functions** coverage types). Here I'm specifically talking about **Branches** coverage type numbers - these should be also not  bad, but with non-empty `CETL_DEBUG_ASSERT` it won't be possible improve them. It's very rarely you could artificially craft environment so, that assert will be failing (and caught by "expected to crash" test). IMHO targetting asserts coverage is too much. But leaving them as is means that we won't detect real lacks of branch coverage in some production business logic. In other words - it's noise for me.

Reply:
ok